src/components/views/voip/VideoLobby.tsx

```diff
@@ -20,115 +20,120 @@ import { logger } from "matrix-js-sdk/src/logger";
+import DeviceContextMenu from "../context_menus/DeviceContextMenu";
 import { Alignment } from "../elements/Tooltip";
 import AccessibleButton from "../elements/AccessibleButton";
 import AccessibleTooltipButton from "../elements/AccessibleTooltipButton";
 import FacePile from "../elements/FacePile";
 import MemberAvatar from "../avatars/MemberAvatar";
 
 interface IDeviceButtonProps {
-    kind: string;
+    kind: MediaDeviceKindEnum;
     devices: MediaDeviceInfo[];
-    setDevice: (device: MediaDeviceInfo) => void;
-    deviceListLabel: string;
     active: boolean;
     disabled: boolean;
     toggle: () => void;
-    activeTitle: string;
-    inactiveTitle: string;
 }
 
-const DeviceButton: FC<IDeviceButtonProps> = ({
-    kind, devices, setDevice, deviceListLabel, active, disabled, toggle, activeTitle, inactiveTitle,
-}) => {
+const DeviceButton: FC<IDeviceButtonProps> = ({ kind, devices, active, disabled, toggle }) => {
     // Depending on permissions, the browser might not let us know device labels,
-    // in which case there's nothing helpful we can display
-    const labelledDevices = useMemo(() => devices.filter(d => d.label.length), [devices]);
+    // in which case the context menu won't display anything helpful
+    const labelledDevices = useMemo(() => devices.filter(d => d.label), [devices]);
 
     const [menuDisplayed, buttonRef, openMenu, closeMenu] = useContextMenu();
-    let contextMenu;
-    if (menuDisplayed) {
-        const selectDevice = (device: MediaDeviceInfo) => {
-            setDevice(device);
-            closeMenu();
-        };
-
-        const buttonRect = buttonRef.current.getBoundingClientRect();
-        contextMenu = <IconizedContextMenu {...aboveLeftOf(buttonRect)} onFinished={closeMenu}>
-            <IconizedContextMenuOptionList>
-                { labelledDevices.map(d =>
-                    <IconizedContextMenuOption
-                        key={d.deviceId}
-                        label={d.label}
-                        onClick={() => selectDevice(d)}
-                    />,
-                ) }
-            </IconizedContextMenuOptionList>
-        </IconizedContextMenu>;
-    }
 
     if (!devices.length) return null;
 
+    let className: string;
+    let title: string;
+    let listTitle: string;
+    let deviceKinds: MediaDeviceKindEnum[];
+    if (kind === MediaDeviceKindEnum.AudioInput) {
+        className = "mx_VideoLobby_deviceButton_audio";
+        title = active ? _t("Mute microphone") : _t("Unmute microphone");
+        listTitle = _t("Audio devices");
+        deviceKinds = [MediaDeviceKindEnum.AudioInput, MediaDeviceKindEnum.AudioOutput];
+    } else if (kind === MediaDeviceKindEnum.VideoInput) {
+        className = "mx_VideoLobby_deviceButton_video";
+        title = active ? _t("Turn off camera") : _t("Turn on camera");
+        listTitle = _t("Video devices");
+        deviceKinds = [MediaDeviceKindEnum.VideoInput];
+    }
```

Comment:
fwiw, if you want to avoid having to overly specify types, keep the code small, and still have developer safety:

```suggestion
    let className = "mx_VideoLobby_deviceButton_audio";
    let title = active ? _t("Mute microphone") : _t("Unmute microphone");
    let listTitle = _t("Audio devices");
    let deviceKinds = [MediaDeviceKindEnum.AudioInput, MediaDeviceKindEnum.AudioOutput];
    if (kind === MediaDeviceKindEnum.VideoInput) {
        className = "mx_VideoLobby_deviceButton_video";
        title = active ? _t("Turn off camera") : _t("Turn on camera");
        listTitle = _t("Video devices");
        deviceKinds = [MediaDeviceKindEnum.VideoInput];
    } else {
        throw new Error(`Unexpected media device kind: ${kind}`);
    }
```

Reply:
My personal preference is for the code that's there now, because it's more explicit about which cases are being handled, and avoids making calls to `_t` that then get immediately discarded, so I'm going to pass on the suggestion as long as that's okay :slightly_smiling_face: